File: app/src/main/java/com/example/apiboutiqueexame/api/Livro.kt
package com.example.apiboutiqueexame.api

import com.google.gson.annotations.SerializedName

data class Livro(
    @SerializedName("_id") val id: String,
    @SerializedName("titulo") val titulo: String,
    @SerializedName("descricao") val descricao: String,
    @SerializedName("editora") val editora: String?,
    @SerializedName("image") val imageUrl: String?,
    @SerializedName("autor") val autor: Autor,
    val userId: Int
)

data class LivroReq(
    val titulo: String,
    val descricao: String,
    val editora: String?,
    val image: String,
    val userId: Int,
    val autor: Autor

)

data class Autor(
    val nome: String,
    val nacionalidade: String
)

File: models/livro.js
const mongoose = require("mongoose");
const autorSchema = require("./autor");

const livroSchema = new mongoose.Schema({
    titulo: { type: String, required: true },
    descricao: { type: String, required: true },
    conteudo: {type: String, require: true},
    editora: { type: String },
    image: { type: String },
    userId: { type: Number },
    autor: autorSchema,  
});

module.exports = mongoose.model("Livro", livroSchema);
